### DISCUSSION TITLE
Can this control only a single pump?  Or does it need an IntelliCenter Controller

### DISCUSSION THREAD

**Discussion by gherlein** [tagyoureit/nodejs-poolController#412] [Nixie] [answered]
I bought a house.  It has a pool.  The pool has a Pentair pump.  I now have an RS-486 adapter connected to it and today can finally start testing.  I'd love to use this software, but I keep seeing that it connects to an IntelliCenter Controller.  Must I have that?  All I really want at this point is to remotely turn my pump on and off at certain times.  That's it.  I am a coder, I can do low-level serial code and can write something if I have to.  But I would rather use something that already works.  Can this code directly control the pump?

**Comment by rstrouse** [comment]
It does not need an IntelliCenter to control a single pump.  It can connect to IntelliCenter, EasyTouch, or IntelliTouch but it also stands alone and creates a controller named Nixie.

**Reply by j4rs** [reply]
Is there a chance to make it work on an IntelliConnect from Pentair? 

Thanks in advance.

**Reply by rstrouse** [reply]
Unfortunately, we have not done any work with IntelliConnect.  The problem with it is that it doesn't do much overall.  There are no valve ports, and only two relays so folks rapidly outgrow the functionality.  

This controller will not work well if you have any of the following equipment.  You might be able to cobble it though or even extend it with a Nixie controller.
1. Solar (no valve control or solar temp sensor)
2. Combined Pool/Spa (no valve control)
3. Heat Pump (the air sensor differential doesn't exist)
4. Valve position water features (no valve control)
5. IntelliFlo VF, XF, VSF (it only speaks the vanilla VS protocol)
6. More than 1 aux control + chlorinator or more than 2 aux control without chlorinator

It does have a single temp sensor, RS485 port, and a heater fireman's switch but beyond that there really aren't enough relays.  If you have a chlorinator hooked up one of those relays gets dedicated to a filter relay leaving only 1 aux relay.  If you are brave I suppose you could rely on the flow switch for the chlorinator.  But that easily fails in the on position if the top of the paddle corrodes where it goes into the plastic housing.

It would be interesting to see what the traffic on the port is though regarding the status of the two relays.  I know what the traffic is for the pumps and chlorinators.

**Comment by gherlein** [comment]
OK, it's installed and it is opening the serial port that has the RS-485 adapter.  

[4/14/2022, 7:56:06 AM] warn: Inactivity timeout for 0 serial port /dev/ttyUSB0 after 10 seconds
[4/14/2022, 7:56:06 AM] info: Serial Port 0 has been closed 0:

I am suspecting I am missing a configuration.  Where is that documented?  Does the pump need to be in a particular mode for this to work?

**Reply by rstrouse** [reply]
njsPC will close and reopen the comm port if it does not detect any activity.  Tell us a little bit about the equipment you have attached to the RS485 port.  If you don't have any RS485 equipment you can simply click disable on the port but if you have any of the following you need to have it enabled.
1. Any Pentair controller: EasyTouch, IntelliTouch, or IntelliCenter
2. Pentair Pumps VS,VSF,VF,XF.
3. Hayward Pumps (Install on an Aux Port)
4. RS485 controlled chlorinators from Pentair or Hayward
5. IntelliChem
6. UltraTemp compatible heat pump

If you don't have any of these items then the port is not important and you can disable it.  If you do have any of the Pentair controllers and are not seeing traffic on RS485 then this is likely a wiring or adapter issue.  If you have any of the other items then you first need to configure them before njsPC will talk to them.  They will not respond unless they have been asked to do so.

You can find a whole lot of information on how to configure a Nixie controller in this wiki
https://github.com/tagyoureit/nodejs-poolController/wiki/Nixie-Controller-Setup

**Comment by tagyoureit** [comment, marked as answer]
We previously tried to interact with IntelliConnect https://github.com/tagyoureit/nodejs-poolController/issues/158 and it seems to be local control only and not outwardly broadcast any status.
